### PR TITLE
Class22 129 switching page buy now button at product page

### PR DESCRIPTION
**Pull request by patel-prakashkumar**
# Description:

# Jira:

### CODE REVIEW

File: packages/client/src/components/ViewDropDown/ViewDropdown.component.js [hunk 1]
```diff
@@ -12,11 +13,19 @@ const ViewDropdown = ({ options, label }) => {
         </option>
       );
     });
+
+  const handleChange = (event) => {
+    setValue(event.target.value);
+    onSelect(event.target.value);
```
Comment (orhantoy) [inline packages/client/src/components/ViewDropDown/ViewDropdown.component.js]: Why are we setting a value and calling a callback? Did you consider making value a prop?

File: packages/client/src/containers/DropdownGroup/DropdownGroup.Container.js [hunk 2]
```diff
@@ -35,21 +35,47 @@ export const DropdownGroup = ({ productId }) => {
...
+          onSelect={(size) => selectedVariantType('size', size)}
+        />
       </div>
+      {/*  <div className="dropdown-group">
+        <ViewDropdown
+          options={quantities}
+          label="Quantity"
+          onSelect={() => {}}
+        />
+      </div> */}
```
Comment (orhantoy) [inline packages/client/src/containers/DropdownGroup/DropdownGroup.Container.js]: Can we delete this?

File: packages/client/src/containers/ProductView/ProductView.container.js [hunk 3]
```diff
@@ -26,7 +29,54 @@ export const ProductView = () => {
...
+    if (variantType === 'size') {
+      setOrderValue((preValue) => {
+        return { ...preValue, size: variantValue };
+      });
+    }
+    if (variantType === 'quantity') {
+      setOrderValue((preValue) => {
+        return { ...preValue, quantity: variantValue };
+      });
+    }
```
Comment (orhantoy) [inline packages/client/src/containers/ProductView/ProductView.container.js]: You can achieve the same thing with `setOrderValue((prev) => ({ ...prev, [variantType]: variantValue }))`.

**Review by orhantoy — COMMENTED** [review]
Looks good! Can you update the PR description and also include a short demo of this working?

File: packages/client/src/containers/DropdownGroup/DropdownGroup.Container.js (hunk 2 above)
Comment (patel-prakashkumar) [reply packages/client/src/containers/DropdownGroup/DropdownGroup.Container.js]: I am waiting to approved and merge PR: https://github.com/HackYourFuture-CPH/fp-class22/pull/138 than I can get quantity from variant table. right now I get hardcoded value as quantity.

File: packages/client/src/components/ViewDropDown/ViewDropdown.component.js (hunk 1 above)
Comment (patel-prakashkumar) [reply packages/client/src/components/ViewDropDown/ViewDropdown.component.js]: I am getting value onSelect function from dropdown and second one updated state and store value.

File: packages/client/src/containers/DropdownGroup/DropdownGroup.Container.js (hunk 2 above)
Comment (patel-prakashkumar) [reply packages/client/src/containers/DropdownGroup/DropdownGroup.Container.js]: I have deleted.

File: packages/client/src/containers/ProductView/ProductView.container.js (hunk 3 above)
Comment (patel-prakashkumar) [reply packages/client/src/containers/ProductView/ProductView.container.js]: I have changed.

**Review by RZhyvitskyi — APPROVED** [review]

**Review by orhantoy — APPROVED** [review]

**Pull request merged**